mypy errors

Since traitlets v5.2.0, mypy errors out with:
```py
from traitlets import Integer

# error: Module "traitlets" has no attribute "Integer"  [attr-defined]
#     from traitlets import Integer
#     ^
# Found 1 error in 1 file (checked 1 source file)
```

--- a/traitlets/tests/test_traitlets.py
+++ b/traitlets/tests/test_traitlets.py
@@ -3137,7 +3137,7 @@ def test_tcp_from_string(s, expected):
 
 
 def test_all_attribute():
-    """Verify all trait types are added to to `traitlets.__all__`"""
+    """Verify all trait types are added to `traitlets.__all__`"""
     names = dir(traitlets)
     for name in names:
         value = getattr(traitlets, name)

--- a/traitlets/tests/test_traitlets.py
+++ b/traitlets/tests/test_traitlets.py
@@ -3137,10 +3137,7 @@ def test_tcp_from_string(s, expected):
 
 
 def test_all_attribute():
-    """verify all trait types are added to to `traitlets.__all__`
-
-    do in a function to avoid iterating through globals while defining local variables
-    """
+    """Verify all trait types are added to to `traitlets.__all__`"""
     names = dir(traitlets)
     for name in names:
         value = getattr(traitlets, name)

--- a/traitlets/tests/test_traitlets.py
+++ b/traitlets/tests/test_traitlets.py
@@ -56,6 +56,7 @@ from traitlets import (
     link,
     observe,
     observe_compat,
+    traitlets,
     validate,
 )
 from traitlets.utils import cast_unicode
@@ -3133,3 +3134,20 @@ def test_object_from_string(s, expected):
 )
 def test_tcp_from_string(s, expected):
     _from_string_test(TCPAddress, s, expected)
+
+
+def test_all_attribute():
+    """verify all trait types are added to to `traitlets.__all__`
+
+    do in a function to avoid iterating through globals while defining local variables
+    """
+    names = dir(traitlets)
+    for name in names:
+        value = getattr(traitlets, name)
+        if not name.startswith("_") and isinstance(value, type) and issubclass(value, TraitType):
+            if name not in traitlets.__all__:
+                raise ValueError(f"{name} not in __all__")
+
+    for name in traitlets.__all__:
+        if name not in names:
+            raise ValueError(f"{name} should be removed from __all__")

--- a/traitlets/traitlets.py
+++ b/traitlets/traitlets.py
@@ -74,6 +74,7 @@ __all__ = [
     "CComplex",
     "CFloat",
     "CInt",
+    "CLong",
     "CRegExp",
     "CUnicode",
     "Callable",
@@ -95,7 +96,9 @@ __all__ = [
     "HasTraits",
     "Instance",
     "Int",
+    "Integer",
     "List",
+    "Long",
     "MetaHasDescriptors",
     "MetaHasTraits",
     "ObjectName",
@@ -3455,16 +3458,3 @@ class Callable(TraitType):
             return value
         else:
             self.error(obj, value)
-
-
-def _add_all():
-    """add all trait types to `__all__`
-
-    do in a function to avoid iterating through globals while defining local variables
-    """
-    for _name, _value in globals().items():
-        if not _name.startswith("_") and isinstance(_value, type) and issubclass(_value, TraitType):
-            __all__.append(_name)
-
-
-_add_all()

